File: server/Records/app/ax_types/observation.py
from typing import Any
import strawberry
from  datetime import datetime
from typing import Optional,List
from app.ax_types.util import Json_type


@strawberry.type
class Observation:
    id:Optional[str] = None
    patientID:Optional[str] = None
    labID:Optional[str] = None
    code:Optional[str] = None
    display:Optional[str] = None
    unit:Optional[str] = None
    value:Optional[str] = None
    timestamp:Optional[str] = None
    meta:Optional[Json_type]= None # type: ignore


@strawberry.type
class ObservationStack:
    Observations :List[Observation] # type: ignore

File: axion/project 3/flutter_application_1/lib/models/observation.dart
class Observation {
  final String? encounter;
  final String patient;
  final String code;
  final String display;
  final String? unit;
  final dynamic value;
  final String timestamp;
  final Map<String, dynamic> meta;

  Observation({
    this.encounter,
    required this.patient,
    required this.code,
    required this.display,
    this.unit,
    required this.value,
    required this.timestamp,
    required this.meta,
  });

  factory Observation.fromJson(Map<String, dynamic> json) {
    return Observation(
      encounter: json['encounter'] as String?,
      patient: json['patient'] as String,
      code: json['code'] as String,
      display: json['display'] as String,
      unit: json['unit'] as String?,
      value: json['value'],
      timestamp: json['timestamp'] as String,
      meta: json['meta'] as Map<String, dynamic>,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'encounter': encounter,
      'patient': patient,
      'code': code,
      'display': display,
      'unit': unit,
      'value': value,
      'timestamp': timestamp,
      'meta': meta,
    };
  }
}
